### ISSUE TITLE
help text for `--denylist` is ambiguous

### ISSUE THREAD

**Issue by grenade** [paritytech/substrate-telemetry#432]
`telemetry_core --help` states that `--denylist` is a "Space delimited list of the names of chains that are not allowed to connect to telemetry. Case sensitive".

this leaves ambiguity around how to specify chain names that contain spaces. presumably:
```
--denylist some_chain "Some Chain"
```
would probably work, but begs the questions:
- why not use the chain id, which doesn't contain spaces?
- where the help says "names of chains", does it really mean "chain id's"?

i currently just specify both the name and id separated by spaces in the hope that it gets figured out internally but i would like to understand what is actually required.

**Comment by dvdplm** [comment]
You are right the docs are not good here. It is the chain id and this should be clarified.

That said, I suspect that using chain ID isn't good enough though as there's nothing stopping two chains from having the same name (and sadly it's not even uncommon for this to occur). The fix for that would be to pass in the genesis hash, which is a bit unwieldy for users.

How about we do this:

- `--denylist` accepts both a chain id (possibly ambiguous) and a genesis hash (`0x`-prefixed hex string)
- any chain that presents a chain id that is on the denylist is rejected
- no spaces or `"`, `'` etc allowed (exits with error)

**Comment by jsdw** [comment]
From our internal chat, Telemetry is only sent the genesis hash and chain name (the thing that can contain spaces).

Soo, I'd be happy with `--denylist` accepting either genesis hash or chain name (the space separated pretty name), and tweaking the docs to make usage clear.
